// @angular/material/menu/typings/index.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
import * as i0 from '@angular/core';
import * as i1 from '@angular/material/menu';
import * as i2 from '@angular/cdk/platform';
import * as i3 from '@angular/cdk/scrolling';
import * as i4 from '@angular/cdk/overlay';
import * as i5 from '@angular/common';
import * as i6 from '@angular/cdk/bidi';
import * as i7 from '@angular/platform-browser';
import * as i8 from '@angular/cdk/portal';
import * as i9 from '@angular/material/core';
export var MatMenuModuleNgFactory = i0.ɵcmf(i1.MatMenuModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(4608, i2.Platform, i2.Platform, []), i0.ɵmpd(5120, i3.ScrollDispatcher, i3.SCROLL_DISPATCHER_PROVIDER_FACTORY, [[3, i3.ScrollDispatcher], i0.NgZone,
            i2.Platform]), i0.ɵmpd(5120, i3.ViewportRuler, i3.VIEWPORT_RULER_PROVIDER_FACTORY, [[3, i3.ViewportRuler], i2.Platform, i0.NgZone, i3.ScrollDispatcher]), i0.ɵmpd(4608, i4.ScrollStrategyOptions, i4.ScrollStrategyOptions, [i3.ScrollDispatcher, i3.ViewportRuler]),
        i0.ɵmpd(5120, i4.OverlayContainer, i4.ɵa, [[3, i4.OverlayContainer]]), i0.ɵmpd(4608, i4.ɵf, i4.ɵf, [i3.ViewportRuler]), i0.ɵmpd(4608, i4.Overlay, i4.Overlay, [i4.ScrollStrategyOptions,
            i4.OverlayContainer, i0.ComponentFactoryResolver, i4.ɵf, i0.ApplicationRef,
            i0.Injector, i0.NgZone]), i0.ɵmpd(5120, i4.ɵc, i4.ɵd, [i4.Overlay]), i0.ɵmpd(4608, i5.NgLocalization, i5.NgLocaleLocalization, [i0.LOCALE_ID]), i0.ɵmpd(6144, i6.DIR_DOCUMENT, null, [i7.DOCUMENT]), i0.ɵmpd(4608, i6.Directionality, i6.Directionality, [[2, i6.DIR_DOCUMENT]]), i0.ɵmpd(5120, i1.MAT_MENU_SCROLL_STRATEGY, i1.ɵc22, [i4.Overlay]), i0.ɵmpd(512, i8.PortalModule, i8.PortalModule, []),
        i0.ɵmpd(512, i2.PlatformModule, i2.PlatformModule, []), i0.ɵmpd(512, i3.ScrollDispatchModule, i3.ScrollDispatchModule, []), i0.ɵmpd(512, i4.OverlayModule, i4.OverlayModule, []), i0.ɵmpd(512, i5.CommonModule, i5.CommonModule, []), i0.ɵmpd(512, i9.CompatibilityModule, i9.CompatibilityModule, []), i0.ɵmpd(512, i6.BidiModule, i6.BidiModule, []),
        i0.ɵmpd(256, i9.MATERIAL_SANITY_CHECKS, true, []), i0.ɵmpd(512, i9.MatCommonModule, i9.MatCommonModule, [[2, i9.MATERIAL_SANITY_CHECKS]]), i0.ɵmpd(512, i9.MatRippleModule, i9.MatRippleModule, []), i0.ɵmpd(512, i1.MatMenuModule, i1.MatMenuModule, []), i0.ɵmpd(256, i1.MAT_MENU_DEFAULT_OPTIONS, { overlapTrigger: true,
            xPosition: 'after', yPosition: 'below' }, [])]);
});
var styles_MatMenu = ['.mat-menu-panel{min-width:112px;max-width:280px;overflow:auto;-webkit-overflow-scrolling:touch;max-height:calc(100vh - 48px);border-radius:2px}.mat-menu-panel:not([class*=mat-elevation-z]){box-shadow:0 3px 1px -2px rgba(0,0,0,.2),0 2px 2px 0 rgba(0,0,0,.14),0 1px 5px 0 rgba(0,0,0,.12)}.mat-menu-panel.mat-menu-after.mat-menu-below{transform-origin:left top}.mat-menu-panel.mat-menu-after.mat-menu-above{transform-origin:left bottom}.mat-menu-panel.mat-menu-before.mat-menu-below{transform-origin:right top}.mat-menu-panel.mat-menu-before.mat-menu-above{transform-origin:right bottom}[dir=rtl] .mat-menu-panel.mat-menu-after.mat-menu-below{transform-origin:right top}[dir=rtl] .mat-menu-panel.mat-menu-after.mat-menu-above{transform-origin:right bottom}[dir=rtl] .mat-menu-panel.mat-menu-before.mat-menu-below{transform-origin:left top}[dir=rtl] .mat-menu-panel.mat-menu-before.mat-menu-above{transform-origin:left bottom}.mat-menu-panel.ng-animating{pointer-events:none}@media screen and (-ms-high-contrast:active){.mat-menu-panel{outline:solid 1px}}.mat-menu-content{padding-top:8px;padding-bottom:8px}.mat-menu-item{-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;cursor:pointer;outline:0;border:none;-webkit-tap-highlight-color:transparent;white-space:nowrap;overflow:hidden;text-overflow:ellipsis;display:block;line-height:48px;height:48px;padding:0 16px;text-align:left;text-decoration:none;position:relative}.mat-menu-item[disabled]{cursor:default}[dir=rtl] .mat-menu-item{text-align:right}.mat-menu-item .mat-icon{margin-right:16px}[dir=rtl] .mat-menu-item .mat-icon{margin-left:16px;margin-right:0}.mat-menu-item .mat-icon{vertical-align:middle}.mat-menu-item-submenu-trigger{padding-right:32px}.mat-menu-item-submenu-trigger::after{width:0;height:0;border-style:solid;border-width:5px 0 5px 5px;border-color:transparent transparent transparent currentColor;content:\'\';display:inline-block;position:absolute;top:50%;right:16px;transform:translateY(-50%)}[dir=rtl] .mat-menu-item-submenu-trigger{padding-right:8px;padding-left:32px}[dir=rtl] .mat-menu-item-submenu-trigger::after{right:auto;left:16px;transform:rotateY(180deg) translateY(-50%)}button.mat-menu-item{width:100%}.mat-menu-ripple{top:0;left:0;right:0;bottom:0;position:absolute}'];
export var RenderType_MatMenu = i0.ɵcrt({ encapsulation: 2, styles: styles_MatMenu,
    data: { 'animation': [{ type: 7, name: 'transformMenu', definitions: [{ type: 0, name: 'void',
                        styles: { type: 6, styles: { opacity: 0, transform: 'scale(0.01, 0.01)' }, offset: null },
                        options: undefined }, { type: 0, name: 'enter-start', styles: { type: 6, styles: { opacity: 1,
                                transform: 'scale(1, 0.5)' }, offset: null }, options: undefined },
                    { type: 0, name: 'enter', styles: { type: 6, styles: { transform: 'scale(1, 1)' }, offset: null },
                        options: undefined }, { type: 1, expr: 'void => enter-start', animation: { type: 4,
                            styles: null, timings: '100ms linear' }, options: null }, { type: 1,
                        expr: 'enter-start => enter', animation: { type: 4, styles: null, timings: '300ms cubic-bezier(0.25, 0.8, 0.25, 1)' },
                        options: null }, { type: 1, expr: '* => void', animation: { type: 4, styles: { type: 6,
                                styles: { opacity: 0 }, offset: null }, timings: '150ms 50ms linear' },
                        options: null }], options: {} }, { type: 7, name: 'fadeInItems', definitions: [{ type: 0,
                        name: 'showing', styles: { type: 6, styles: { opacity: 1 }, offset: null }, options: undefined },
                    { type: 1, expr: 'void => *', animation: [{ type: 6, styles: { opacity: 0 }, offset: null },
                            { type: 4, styles: null, timings: '400ms 100ms cubic-bezier(0.55, 0, 0.55, 0.2)' }],
                        options: null }], options: {} }] } });
function View_MatMenu_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 3, 'div', [['class',
                'mat-menu-panel'], ['role', 'menu']], [[24, '@transformMenu', 0]], [[null,
                'keydown'], [null, 'click'], [null, '@transformMenu.done']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('keydown' === en)) {
                var pd_0 = (_co._handleKeydown($event) !== false);
                ad = (pd_0 && ad);
            }
            if (('click' === en)) {
                var pd_1 = (_co.close.emit('click') !== false);
                ad = (pd_1 && ad);
            }
            if (('@transformMenu.done' === en)) {
                var pd_2 = (_co._onAnimationDone($event) !== false);
                ad = (pd_2 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(278528, null, 0, i5.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), (_l()(), i0.ɵeld(0, null, null, 1, 'div', [['class', 'mat-menu-content']], [[24, '@fadeInItems', 0]], null, null, null, null)), i0.ɵncd(null, 0)], function (_ck, _v) {
        var _co = _v.component;
        var currVal_1 = 'mat-menu-panel';
        var currVal_2 = _co._classList;
        _ck(_v, 1, 0, currVal_1, currVal_2);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co._panelAnimationState;
        _ck(_v, 0, 0, currVal_0);
        var currVal_3 = 'showing';
        _ck(_v, 2, 0, currVal_3);
    });
}
export function View_MatMenu_0(_l) {
    return i0.ɵvid(2, [i0.ɵqud(402653184, 1, { templateRef: 0 }), (_l()(), i0.ɵand(0, [[1, 2]], null, 0, null, View_MatMenu_1))], null, null);
}
export function View_MatMenu_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 2, 'mat-menu', [], null, null, null, View_MatMenu_0, RenderType_MatMenu)),
        i0.ɵdid(1228800, null, 1, i1.MatMenu, [i0.ElementRef, i0.NgZone, i1.MAT_MENU_DEFAULT_OPTIONS], null, null), i0.ɵqud(603979776, 1, { items: 1 })], null, null);
}
export var MatMenuNgFactory = i0.ɵccf('mat-menu', i1.MatMenu, View_MatMenu_Host_0, { xPosition: 'xPosition', yPosition: 'yPosition', overlapTrigger: 'overlapTrigger',
    classList: 'class' }, { close: 'close' }, ['*']);
var styles_MatMenuItem = [];
export var RenderType_MatMenuItem = i0.ɵcrt({ encapsulation: 2, styles: styles_MatMenuItem,
    data: {} });
function View_MatMenuItem_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'div', [['class',
                'mat-menu-ripple mat-ripple'], ['mat-ripple', '']], [[2, 'mat-ripple-unbounded',
                null]], null, null, null, null)), i0.ɵdid(671744, null, 0, i9.MatRipple, [i0.ElementRef, i0.NgZone, i2.Platform, [2, i9.MAT_RIPPLE_GLOBAL_OPTIONS]], { trigger: [0, 'trigger'] }, null)], function (_ck, _v) {
        var _co = _v.component;
        var currVal_1 = _co._getHostElement();
        _ck(_v, 1, 0, currVal_1);
    }, function (_ck, _v) {
        var currVal_0 = i0.ɵnov(_v, 1).unbounded;
        _ck(_v, 0, 0, currVal_0);
    });
}
export function View_MatMenuItem_0(_l) {
    return i0.ɵvid(2, [i0.ɵncd(null, 0), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_MatMenuItem_1)), i0.ɵdid(16384, null, 0, i5.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null)], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = !_co.disabled;
        _ck(_v, 2, 0, currVal_0);
    }, null);
}
export function View_MatMenuItem_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'div', [['class',
                'mat-menu-item'], ['mat-menu-item', ''], ['role', 'menuitem']], [[2, 'mat-menu-item-highlighted',
                null], [2, 'mat-menu-item-submenu-trigger', null], [1, 'tabindex',
                0], [1, 'aria-disabled', 0], [1, 'disabled', 0]], [[null, 'click'], [null,
                'mouseenter']], function (_v, en, $event) {
            var ad = true;
            if (('click' === en)) {
                var pd_0 = (i0.ɵnov(_v, 1)._checkDisabled($event) !== false);
                ad = (pd_0 && ad);
            }
            if (('mouseenter' === en)) {
                var pd_1 = (i0.ɵnov(_v, 1)._emitHoverEvent() !== false);
                ad = (pd_1 && ad);
            }
            return ad;
        }, View_MatMenuItem_0, RenderType_MatMenuItem)), i0.ɵdid(180224, null, 0, i1.MatMenuItem, [i0.ElementRef], null, null)], null, function (_ck, _v) {
        var currVal_0 = i0.ɵnov(_v, 1)._highlighted;
        var currVal_1 = i0.ɵnov(_v, 1)._triggersSubmenu;
        var currVal_2 = i0.ɵnov(_v, 1)._getTabIndex();
        var currVal_3 = i0.ɵnov(_v, 1).disabled.toString();
        var currVal_4 = (i0.ɵnov(_v, 1).disabled || null);
        _ck(_v, 0, 0, currVal_0, currVal_1, currVal_2, currVal_3, currVal_4);
    });
}
export var MatMenuItemNgFactory = i0.ɵccf('[mat-menu-item]', i1.MatMenuItem, View_MatMenuItem_Host_0, { disabled: 'disabled' }, {}, ['*']);
