// @angular/material/core/typings/index.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
import * as i0 from '@angular/core';
import * as i1 from '@angular/material/core';
import * as i2 from '@angular/cdk/bidi';
import * as i3 from '@angular/platform-browser';
import * as i4 from '@angular/cdk/platform';
import * as i5 from '@angular/common';
export var MatCommonModuleNgFactory = i0.ɵcmf(i1.MatCommonModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(6144, i2.DIR_DOCUMENT, null, [i3.DOCUMENT]), i0.ɵmpd(4608, i2.Directionality, i2.Directionality, [[2, i2.DIR_DOCUMENT]]), i0.ɵmpd(512, i1.CompatibilityModule, i1.CompatibilityModule, []), i0.ɵmpd(512, i2.BidiModule, i2.BidiModule, []), i0.ɵmpd(256, i1.MATERIAL_SANITY_CHECKS, true, []),
        i0.ɵmpd(512, i1.MatCommonModule, i1.MatCommonModule, [[2, i1.MATERIAL_SANITY_CHECKS]])]);
});
export var CompatibilityModuleNgFactory = i0.ɵcmf(i1.CompatibilityModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(512, i1.CompatibilityModule, i1.CompatibilityModule, [])]);
});
export var NoConflictStyleCompatibilityModeNgFactory = i0.ɵcmf(i1.NoConflictStyleCompatibilityMode, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(512, i1.NoConflictStyleCompatibilityMode, i1.NoConflictStyleCompatibilityMode, [])]);
});
export var NativeDateModuleNgFactory = i0.ɵcmf(i1.NativeDateModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(6144, i1.MAT_DATE_LOCALE, null, [i0.LOCALE_ID]), i0.ɵmpd(4608, i1.DateAdapter, i1.NativeDateAdapter, [[2, i1.MAT_DATE_LOCALE]]), i0.ɵmpd(512, i1.NativeDateModule, i1.NativeDateModule, [])]);
});
export var MatNativeDateModuleNgFactory = i0.ɵcmf(i1.MatNativeDateModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(6144, i1.MAT_DATE_LOCALE, null, [i0.LOCALE_ID]), i0.ɵmpd(4608, i1.DateAdapter, i1.NativeDateAdapter, [[2, i1.MAT_DATE_LOCALE]]), i0.ɵmpd(512, i1.NativeDateModule, i1.NativeDateModule, []), i0.ɵmpd(512, i1.MatNativeDateModule, i1.MatNativeDateModule, []), i0.ɵmpd(256, i1.MAT_DATE_FORMATS, i1.MAT_NATIVE_DATE_FORMATS, [])]);
});
export var MatLineModuleNgFactory = i0.ɵcmf(i1.MatLineModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(6144, i2.DIR_DOCUMENT, null, [i3.DOCUMENT]), i0.ɵmpd(4608, i2.Directionality, i2.Directionality, [[2, i2.DIR_DOCUMENT]]), i0.ɵmpd(512, i1.CompatibilityModule, i1.CompatibilityModule, []), i0.ɵmpd(512, i2.BidiModule, i2.BidiModule, []), i0.ɵmpd(256, i1.MATERIAL_SANITY_CHECKS, true, []),
        i0.ɵmpd(512, i1.MatCommonModule, i1.MatCommonModule, [[2, i1.MATERIAL_SANITY_CHECKS]]),
        i0.ɵmpd(512, i1.MatLineModule, i1.MatLineModule, [])]);
});
export var MatOptionModuleNgFactory = i0.ɵcmf(i1.MatOptionModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(6144, i2.DIR_DOCUMENT, null, [i3.DOCUMENT]), i0.ɵmpd(4608, i2.Directionality, i2.Directionality, [[2, i2.DIR_DOCUMENT]]), i0.ɵmpd(4608, i4.Platform, i4.Platform, []), i0.ɵmpd(4608, i5.NgLocalization, i5.NgLocaleLocalization, [i0.LOCALE_ID]),
        i0.ɵmpd(512, i1.CompatibilityModule, i1.CompatibilityModule, []),
        i0.ɵmpd(512, i2.BidiModule, i2.BidiModule, []), i0.ɵmpd(256, i1.MATERIAL_SANITY_CHECKS, true, []), i0.ɵmpd(512, i1.MatCommonModule, i1.MatCommonModule, [[2, i1.MATERIAL_SANITY_CHECKS]]), i0.ɵmpd(512, i4.PlatformModule, i4.PlatformModule, []), i0.ɵmpd(512, i1.MatRippleModule, i1.MatRippleModule, []),
        i0.ɵmpd(512, i5.CommonModule, i5.CommonModule, []), i0.ɵmpd(512, i1.MatPseudoCheckboxModule, i1.MatPseudoCheckboxModule, []), i0.ɵmpd(512, i1.MatOptionModule, i1.MatOptionModule, [])]);
});
export var MatRippleModuleNgFactory = i0.ɵcmf(i1.MatRippleModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(6144, i2.DIR_DOCUMENT, null, [i3.DOCUMENT]), i0.ɵmpd(4608, i2.Directionality, i2.Directionality, [[2, i2.DIR_DOCUMENT]]), i0.ɵmpd(4608, i4.Platform, i4.Platform, []), i0.ɵmpd(512, i1.CompatibilityModule, i1.CompatibilityModule, []), i0.ɵmpd(512, i2.BidiModule, i2.BidiModule, []), i0.ɵmpd(256, i1.MATERIAL_SANITY_CHECKS, true, []), i0.ɵmpd(512, i1.MatCommonModule, i1.MatCommonModule, [[2, i1.MATERIAL_SANITY_CHECKS]]), i0.ɵmpd(512, i4.PlatformModule, i4.PlatformModule, []), i0.ɵmpd(512, i1.MatRippleModule, i1.MatRippleModule, [])]);
});
export var MatPseudoCheckboxModuleNgFactory = i0.ɵcmf(i1.MatPseudoCheckboxModule, [], function (_l) {
    return i0.ɵmod([i0.ɵmpd(512, i0.ComponentFactoryResolver, i0.ɵCodegenComponentFactoryResolver, [[8, []], [3, i0.ComponentFactoryResolver], i0.NgModuleRef]), i0.ɵmpd(512, i1.MatPseudoCheckboxModule, i1.MatPseudoCheckboxModule, [])]);
});
var styles_MatOption = [];
export var RenderType_MatOption = i0.ɵcrt({ encapsulation: 2, styles: styles_MatOption,
    data: {} });
function View_MatOption_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 3, 'span', [], null, null, null, null, null)), (_l()(),
            i0.ɵeld(0, null, null, 2, 'mat-pseudo-checkbox', [['class', 'mat-option-pseudo-checkbox mat-pseudo-checkbox']], [[2, 'mat-pseudo-checkbox-indeterminate', null], [2, 'mat-pseudo-checkbox-checked',
                    null], [2, 'mat-pseudo-checkbox-disabled', null]], null, null, View_MatPseudoCheckbox_0, RenderType_MatPseudoCheckbox)), i0.ɵdid(16384, null, 0, i1.MatPrefixRejector, [], null, null),
        i0.ɵdid(49152, null, 0, i1.MatPseudoCheckbox, [], { state: [0, 'state'],
            disabled: [1, 'disabled'] }, null)], function (_ck, _v) {
        var _co = _v.component;
        var currVal_3 = (_co.selected ? 'checked' : '');
        var currVal_4 = _co.disabled;
        _ck(_v, 3, 0, currVal_3, currVal_4);
    }, function (_ck, _v) {
        var currVal_0 = (i0.ɵnov(_v, 3).state === 'indeterminate');
        var currVal_1 = (i0.ɵnov(_v, 3).state === 'checked');
        var currVal_2 = i0.ɵnov(_v, 3).disabled;
        _ck(_v, 1, 0, currVal_0, currVal_1, currVal_2);
    });
}
export function View_MatOption_0(_l) {
    return i0.ɵvid(2, [(_l()(), i0.ɵand(16777216, null, null, 1, null, View_MatOption_1)), i0.ɵdid(16384, null, 0, i5.NgIf, [i0.ViewContainerRef,
            i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵeld(0, null, null, 1, 'span', [['class', 'mat-option-text']], null, null, null, null, null)), i0.ɵncd(null, 0), (_l()(),
            i0.ɵeld(0, null, null, 1, 'div', [['class', 'mat-option-ripple mat-ripple'],
                ['mat-ripple', '']], [[2, 'mat-ripple-unbounded', null]], null, null, null, null)), i0.ɵdid(671744, null, 0, i1.MatRipple, [i0.ElementRef, i0.NgZone, i4.Platform, [2, i1.MAT_RIPPLE_GLOBAL_OPTIONS]], { trigger: [0, 'trigger'], disabled: [1, 'disabled'] }, null)], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.multiple;
        _ck(_v, 1, 0, currVal_0);
        var currVal_2 = _co._getHostElement();
        var currVal_3 = (_co.disabled || _co.disableRipple);
        _ck(_v, 5, 0, currVal_2, currVal_3);
    }, function (_ck, _v) {
        var currVal_1 = i0.ɵnov(_v, 5).unbounded;
        _ck(_v, 4, 0, currVal_1);
    });
}
export function View_MatOption_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'mat-option', [['class',
                'mat-option'], ['role', 'option']], [[1, 'tabindex', 0], [2, 'mat-selected', null],
            [2, 'mat-option-multiple', null], [2, 'mat-active', null], [8, 'id',
                0], [1, 'aria-selected', 0], [1, 'aria-disabled', 0], [2, 'mat-option-disabled',
                null]], [[null, 'click'], [null, 'keydown']], function (_v, en, $event) {
            var ad = true;
            if (('click' === en)) {
                var pd_0 = (i0.ɵnov(_v, 1)._selectViaInteraction() !== false);
                ad = (pd_0 && ad);
            }
            if (('keydown' === en)) {
                var pd_1 = (i0.ɵnov(_v, 1)._handleKeydown($event) !== false);
                ad = (pd_1 && ad);
            }
            return ad;
        }, View_MatOption_0, RenderType_MatOption)), i0.ɵdid(49152, null, 0, i1.MatOption, [i0.ElementRef, i0.ChangeDetectorRef, [2, i1.MatOptgroup]], null, null)], null, function (_ck, _v) {
        var currVal_0 = i0.ɵnov(_v, 1)._getTabIndex();
        var currVal_1 = i0.ɵnov(_v, 1).selected;
        var currVal_2 = i0.ɵnov(_v, 1).multiple;
        var currVal_3 = i0.ɵnov(_v, 1).active;
        var currVal_4 = i0.ɵnov(_v, 1).id;
        var currVal_5 = i0.ɵnov(_v, 1).selected.toString();
        var currVal_6 = i0.ɵnov(_v, 1).disabled.toString();
        var currVal_7 = i0.ɵnov(_v, 1).disabled;
        _ck(_v, 0, 0, currVal_0, currVal_1, currVal_2, currVal_3, currVal_4, currVal_5, currVal_6, currVal_7);
    });
}
export var MatOptionNgFactory = i0.ɵccf('mat-option', i1.MatOption, View_MatOption_Host_0, { value: 'value', disabled: 'disabled' }, { onSelectionChange: 'onSelectionChange' }, ['*']);
var styles_MatOptgroup = [];
export var RenderType_MatOptgroup = i0.ɵcrt({ encapsulation: 2, styles: styles_MatOptgroup,
    data: {} });
export function View_MatOptgroup_0(_l) {
    return i0.ɵvid(2, [(_l()(), i0.ɵeld(0, null, null, 1, 'label', [['class',
                'mat-optgroup-label']], [[8, 'id', 0]], null, null, null, null)), (_l()(), i0.ɵted(null, ['', ''])), i0.ɵncd(null, 0)], null, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co._labelId;
        _ck(_v, 0, 0, currVal_0);
        var currVal_1 = _co.label;
        _ck(_v, 1, 0, currVal_1);
    });
}
export function View_MatOptgroup_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'mat-optgroup', [['class', 'mat-optgroup'], ['role', 'group']], [[2, 'mat-optgroup-disabled', null],
            [1, 'aria-disabled', 0], [1, 'aria-labelledby', 0]], null, null, View_MatOptgroup_0, RenderType_MatOptgroup)), i0.ɵdid(49152, null, 0, i1.MatOptgroup, [], null, null)], null, function (_ck, _v) {
        var currVal_0 = i0.ɵnov(_v, 1).disabled;
        var currVal_1 = i0.ɵnov(_v, 1).disabled.toString();
        var currVal_2 = i0.ɵnov(_v, 1)._labelId;
        _ck(_v, 0, 0, currVal_0, currVal_1, currVal_2);
    });
}
export var MatOptgroupNgFactory = i0.ɵccf('mat-optgroup', i1.MatOptgroup, View_MatOptgroup_Host_0, { disabled: 'disabled', label: 'label' }, {}, ['mat-option']);
var styles_MatPseudoCheckbox = ['.mat-pseudo-checkbox{width:20px;height:20px;border:2px solid;border-radius:2px;cursor:pointer;display:inline-block;vertical-align:middle;box-sizing:border-box;position:relative;flex-shrink:0;transition:border-color 90ms cubic-bezier(0,0,.2,.1),background-color 90ms cubic-bezier(0,0,.2,.1)}.mat-pseudo-checkbox::after{position:absolute;opacity:0;content:\'\';border-bottom:2px solid currentColor;transition:opacity 90ms cubic-bezier(0,0,.2,.1)}.mat-pseudo-checkbox.mat-pseudo-checkbox-checked,.mat-pseudo-checkbox.mat-pseudo-checkbox-indeterminate{border:none}.mat-pseudo-checkbox-disabled{cursor:default}.mat-pseudo-checkbox-indeterminate::after{top:9px;left:2px;width:16px;opacity:1}.mat-pseudo-checkbox-checked::after{top:5px;left:3px;width:12px;height:5px;border-left:2px solid currentColor;transform:rotate(-45deg);opacity:1}'];
export var RenderType_MatPseudoCheckbox = i0.ɵcrt({ encapsulation: 2,
    styles: styles_MatPseudoCheckbox, data: {} });
export function View_MatPseudoCheckbox_0(_l) {
    return i0.ɵvid(2, [], null, null);
}
export function View_MatPseudoCheckbox_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, null, null, 1, 'mat-pseudo-checkbox', [['class', 'mat-pseudo-checkbox']], [[2, 'mat-pseudo-checkbox-indeterminate', null],
            [2, 'mat-pseudo-checkbox-checked', null], [2, 'mat-pseudo-checkbox-disabled',
                null]], null, null, View_MatPseudoCheckbox_0, RenderType_MatPseudoCheckbox)), i0.ɵdid(49152, null, 0, i1.MatPseudoCheckbox, [], null, null)], null, function (_ck, _v) {
        var currVal_0 = (i0.ɵnov(_v, 1).state === 'indeterminate');
        var currVal_1 = (i0.ɵnov(_v, 1).state === 'checked');
        var currVal_2 = i0.ɵnov(_v, 1).disabled;
        _ck(_v, 0, 0, currVal_0, currVal_1, currVal_2);
    });
}
export var MatPseudoCheckboxNgFactory = i0.ɵccf('mat-pseudo-checkbox', i1.MatPseudoCheckbox, View_MatPseudoCheckbox_Host_0, { state: 'state', disabled: 'disabled' }, {}, []);
